perf/kernbench.py

```diff
@@ -0,0 +1,142 @@
+
+import os
+import multiprocessing
+import logging
+import re
+import platform
+
+from avocado import Test
+from avocado import main
+from avocado.utils import build
+from avocado.utils import process
+from avocado.utils import kernel
+from avocado.utils.software_manager import SoftwareManager
+
+log = logging.getLogger('avocado.test')
+
+
+class KernbenchTest(Test):
+    """
+    Kernbench comples the kernel source fetched from the kernel.org
+    and compile it as per the configuration file. Performance figures will be
+    shown in the log file.
+    """
+
+    def time_build(self, kern,
+                   threads=None, timefile=None, make_opts=None):
+        """time the bulding of the kernel"""
+        os.chdir(kern.linux_dir)
+        build.make(kern.linux_dir, extra_args='clean')
+        if kern.config_path is None:
+            build.make(kern.linux_dir, extra_args='defconfig')
+        else:
+            build.make(kern.linux_dir, extra_args='oldnoconfig')
+        process.system('make silentoldconfig')
+        if make_opts:
+            build_string = ("/usr/bin/time -o %s make %s -j %s vmlinux" %
+                            (timefile, make_opts, threads))
+        else:
+            build_string = ("/usr/bin/time -o %s make -j %s vmlinux" %
+                            (timefile, threads))
+        process.system(build_string, allow_output_check='None')
+        if not os.path.isfile('vmlinux'):
+            self.error("No vmlinux found, kernel build failed")
+
+    def setUp(self):
+        """
+        setting up the env for the kernel building
+        """
+        smg = SoftwareManager()
+        deps = ['gcc', 'make', 'automake', 'autoconf', 'time']
+        for package in deps:
+            if smg.check_installed(package) and not smg.install(package):
+                self.error(package + ' is needed for the test to be run')
+
+    def to_seconds(self, time_string):
+        """Converts a string in M+:SS.SS format to S+.SS"""
+        elts = time_string.split(':')
+        if len(elts) == 1:
+            return time_string
+        return str(int(elts[0]) * 60 + float(elts[1]))
+
+    def extract_all_time_results(self, results_string):
+        """Extract user, system, and elapsed times into a list of tuples"""
+        pattern = re.compile(r"(.*?)user (.*?)system (.*?)elapsed")
+        results = []
+        for result in pattern.findall(results_string):
+            results.append(tuple([self.to_seconds(elt) for elt in result]))
+        return results
+
+    def test(self):
+        """
+        Kernel build Test
+        """
+        iterations = self.params.get('runs', default=1)
+        threads = self.params.get('cpus', default=None)
+        version = self.params.get('version', default='4.6')
+        kernel_version = platform.uname()[2]
+        config_path = '/boot/config-' + kernel_version
+        if iterations is None:
+            # Setting the default iteration as one
+            iterations = 1
```

Comment:
this is not needed, as [https://github.com/avocado-framework-tests/avocado-misc-tests/pull/35/files#diff-c481269abe19849c8758d2b1522f49c8R92](url) takes care of this